dkg/pedersen/json/mod.go

```diff
@@ -221,6 +250,60 @@ func (f msgFormat) Encode(ctx serde.Context, msg serde.Message) ([]byte, error)
 		}
 
 		m = Message{DecryptRequest: &req}
+
+	case types.VerifiableDecryptRequest:
+		chiphertexts := in.GetCiphertexts()
+		var encodedCiphertexts []Ciphertext
+
+		for _, cp := range chiphertexts {
+			K, err := cp.K.MarshalBinary()
+			if err != nil {
+				return nil, xerrors.Errorf("couldn't marshal K: %v", err)
+			}
+
+			C, err := cp.C.MarshalBinary()
+			if err != nil {
+				return nil, xerrors.Errorf("couldn't marshal C: %v", err)
+			}
+
+			UBar, err := cp.UBar.MarshalBinary()
+			if err != nil {
+				return nil, xerrors.Errorf("couldn't marshal UBar: %v", err)
+			}
+
+			E, err := cp.E.MarshalBinary()
+			if err != nil {
+				return nil, xerrors.Errorf("couldn't marshal E: %v", err)
+			}
+
+			F, err := cp.F.MarshalBinary()
+			if err != nil {
+				return nil, xerrors.Errorf("couldn't marshal F: %v", err)
+			}
+
+			GBar, err := cp.GBar.MarshalBinary()
+			if err != nil {
```

Comment:
Heh - I love rust's `?` operator :)

Reply:
that's the price for nice errors :)
(to keep in mind: extra clever lines of code are free and not deducted from salary !)